### ISSUE TITLE
Unable to connect to the log-in server on occasions since the Coherent UI swap over.

### ISSUE THREAD

**Issue by aewrynsora** [Healix/Gw2Launcher#300]
I've been following along in the #284 thread but none of the issues listed quite match mine. Ever since the patch that swapped it over, my log-ins are very hit and miss. Some log-in first time, some hang for about 25 seconds and then log-in and some just fail to log-in. I don't know if its something my end, or if something has gone wrong somewhere but it just feels strange that it's more or less been going wrong since that swap over.

I have a log of it below, but its perplexing as it just ends with login successful, whereas a log I had before when this went wrong, it mentioned about a coherent:error as well at the end. Any help/insight would be appreciated, even if it is just letting me know its an issue my end. As I say, just feels very coincidental time-frame wise. 

Thanks in advance!

[00:19:25]: Build 7084 on 10.0.19045.0
[00:19:25]: PID 21656
[00:19:25]: 0 active
[00:19:25]: Logging enabled
[00:19:28][10]: Queued for Launch
[00:19:34][1]: Launching [\Guild Wars 2\Gw2-64.exe] (UpdateVisible)
[00:19:34][1]: Starting process [-image -l:id:1]
[00:19:34][1]: PID 20188 (from 21656) [\Guild Wars 2\Gw2-64.exe] [ -image -l:id:1]
[00:19:36][1]: Launch completed
[00:19:37][1]: Process 20188 exited [00:00:03.3682167] (0)
[00:19:38]: Running scan with options None
[00:19:38][10]: Launching [\Guild Wars 2\Gw2-64.exe] (Launch)
[00:19:38][10]: Starting via proxy (shortcut/none
[00:19:39][10]: PID 24720 (from 24264) [\Guild Wars 2\Gw2-64.exe] [ -windowed -nosound -maploadinfo -shareArchive -l:id:10]
[00:19:40][10]: LauncherWindowHandleCreated
[00:19:40][10]: HandleChanged
[00:19:40][10]: TitleChanged
[00:19:40][10]: Waiting for host process...
[00:19:40][10]: Using mumble link "MumbleLink" with PID 24720
[00:19:40][10]: Delaying until state 4
[00:19:40][10]: CefHost (24952) [\cef\CefHost.exe]
[00:19:40][10]: Found CEF host process
[00:19:40][10]: Waiting for host process (1)
[00:19:40][10]: Waiting for renderer under PID 24720
[00:19:41][10]: Found renderer (18748, 24164)
[00:19:41][10]: Waiting for files to load
[00:19:41][10]: Waiting for files to load... OK
[00:19:41][10]: LauncherWindowLoaded
[00:19:41][10]: Beginning login entry for US type at 31982464 (1120 x 976)
[00:19:42][10]: Entering email using Clipboard
[00:19:42][10]: State 2
[00:19:42][10]: Entering password using Post
[00:19:43][10]: Verifying login
[00:19:48][10]: Login entry complete

**Comment by Healix** [comment]
It sounds like the login entry is succeeding and your problem is entirely with GW2 connecting? When you say it hangs for 25 seconds, you mean it's trying to connect with the circle spinning next to the login button?

That is likely GW2's limiter. They limit how quickly you login per IP, as well as the total number of accounts you can login to per hour. The faster you try to login, the longer it takes until it fails after 60 seconds. You have to stop trying to login for a minute when this happens, otherwise you'll delay all logins indefinitely until you stop. This also applies to website logins. There's a limiter (Settings > Launching > Limiter) to try to avoid this, but if you have a lot of accounts (20+), it can't be avoided without changing your IP. If you're using SMS, they have another limiter on that, even if you don't need the code.

That said, CEF does have problems. Add -usecoherent to the arguments (Settings > Guild Wars 2 > Launch options) to try CoherentUI.

**Comment by Healix** [comment]
Did you add that before or after the log? The log shows you were using CEF, so either you've added it after, or added it to a specific account (Edit > Launch options) instead of all accounts (Settings > Guild Wars 2 > Launch options).

If you want to verify, you can look for CoherentUI or CefHost in the log.

Note the limiter applies to everyone on the IP, so if you happen to be sharing an IP with other people playing GW2, you're all affected.

The only other reason for it to fail to connect is if it can't actually connect. You could try manually selecting a login server, but I doubt that'd help. You can check the servers by right clicking any account > Tools > Support > Servers / Authentication > Search. Picking one of the servers will add it to the launch arguments (-authsrv) for you. Remember to remove it later, as this will eventually cause it to always fail when the IP changes.

**Comment by aewrynsora** [comment]
I did have it added beforehand, then I removed it again and either way I was having the login troubles. I should be the only user right now on my IP logging into gw2 so that bit I'm not too sure on.

The only thing I can say is, I feel it could be coincidentally my end. my ISP I believe pushed out an update around the same time as my internet/hub did reset itself and since then the internet has been all over the place so despite it being around the time of the UI swap over, I feel it might just be my end. I think for now, given noting has really changed and I'm having the connection issues with and without the coherentui argument, Ill just close this for now as I don't believe there is anything extra you'd be able to do.

Thanks for all the help anyway, I'll just hope it rectifies itself, whatever **it** is xD
